instrumentation/bunny/lib/opentelemetry/instrumentation/bunny/patches/channel.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module Bunny
+      module Patches
+        # The Channel module contains the instrumentation patch the Channel#basic_get, Channel#basic_publish and Channel#handle_frameset methods
+        module Channel
+          def basic_get(queue, opts = { manual_ack: false })
+            delivery_info, properties, payload = super
+
+            OpenTelemetry::Instrumentation::Bunny::PatchHelpers.with_receive_span(self, tracer, delivery_info, properties) do
+              properties[:headers] ||= {}
+              OpenTelemetry.propagation.inject(properties[:headers])
+            end
+
+            [delivery_info, properties, payload]
+          end
+
+          def basic_publish(payload, exchange, routing_key, opts = {})
+            OpenTelemetry::Instrumentation::Bunny::PatchHelpers.with_send_span(self, tracer, exchange, routing_key) do
+              opts[:headers] ||= {}
+              OpenTelemetry.propagation.inject(opts[:headers])
+              super(payload, exchange, routing_key, opts)
+            end
+          end
+
+          # This method is called when rabbitmq pushes messages to subscribed consumers
+          def handle_frameset(basic_deliver, properties, content)
+            OpenTelemetry::Instrumentation::Bunny::PatchHelpers.with_receive_span(self, tracer, basic_deliver, properties) do
+              properties[:headers] ||= {}
+              OpenTelemetry.propagation.inject(properties[:headers])
+            end
+
+            super
```

Comment:
Similar to the above, it looks like a [uncaught_exception_handler](https://github.com/ruby-amqp/bunny/blob/master/lib/bunny/channel.rb#L1708) could be provided that raises the exception, in which case we would not capture that event here either.

Not being familiar with rabbitmq, is this call to `super` intentionally not being wrapped with the trace block for any particular reason?

Reply:
`super` only enqueues the message into the internal worker pool in this case and the actual consumer call is wrapped into a process span, which should handle the exception in this case.